traffic_mirroring/app_helper.py

```diff
@@ -182,111 +180,71 @@ def find_matching_tags(instance_tags, config_tags):
+    else:
+        log.info("Creating a session with source: %s, target: %s, filter: %s", 
+                network_interface_id, target_id, filter_id)
+    
+        try:
+            response = create_traffic_mirror_session(ec2, network_interface_id, target_id, filter_id)
+            log.info("Successfully created a traffic mirror session: %s", response)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'TrafficMirrorSourcesPerTargetLimitExceeded':
+                log.info("Target %s has reached its limit.", target_id)
+                raise e
             else:
                 raise e
-        else:
-            raise e
 
 def create_traffic_mirror_session(ec2, network_interface_id, target_id, filter_id):
     return ec2.create_traffic_mirror_session(NetworkInterfaceId=network_interface_id,
                                              TrafficMirrorTargetId=target_id,
                                              TrafficMirrorFilterId=filter_id,
                                              SessionNumber=SESSION_NUMBER)
 
-# Gets or creates a valid target to use in a session. If a targetId was specified in the config
-# it will be used. If not, an existing target will be looked up using tags. If no target is found, one will be created
-def get_or_create_target(ec2, instance, config):
-    if using_predefined_target(config):
-        targetId = config['targetId']
+# Gets a valid target to use in a session. If a targetId was specified in the config
+# it will be used. If not, it will error out with details in the logfile.
+def get_target(instance, config):
+    # customized to retrieve targetID from input config file based on AZ ID of the instance
+    azid = instance.az_id
+    aznum = azid[-3:]    
+    target_name ='targetID-' + aznum
+
+    if field_missing(target_name, config):
+        log.error("TargetID %s not found in config. Please review config yaml file and update targetID.", target_name)
+        return "invalid_target_id"
+    else:
+        targetId = config[target_name]
         log.info('Using provided target: %s', targetId)
         return targetId
 
-    log.info('No target provided in config, determining if one needs to be created')
-    target_subnet_id = get_target_subnet_id(instance, config)
-    target_tag_filter = create_response_filter(
-        "tag:" + SUBNET_TAG_KEY, target_subnet_id)
-    response = describe_targets(ec2, target_tag_filter)
+def get_filter(instance, config):
+    for tag in instance.tags:
+        if tag['Key'] == FILTER_TAG_KEY:
+            if tag['Value'] in ('web','app','db'):
+                filter_name = 'filterID-' + tag['Value']
+                if field_missing(filter_name, config):
+                    log.error("FilterID %s not found in config. Please review config yaml file and update approproate filterID.",filter_name)
+                    return "invalid_filter_id"
```

Comment:
Same comment as in other places, either throw an exception here or return a `None` and handle it instead of an arbitrary string.

Reply:
Made the suggested change and committed updated code.